### PR TITLE
[Buttons] backgroundColorForState: should fall-back to .normal.

### LINKED ISSUE #3411: [Buttons] Unset `backgroundColorForState:` reverts to system color, not .normal color value
### Overview
MDCButton's `backgroundColorForState:` will return a nil color if a state is not explicitly set. This may be intentional, but does not match the behavior of `titleForState:`, `titleColorForState:`, `titleShadowColorForState:` nor `attributedTitleForState:`.  I think it will be more expected for clients if the background color defaults to the value for the `.normal` state to match the related properties.

### Reproduction steps

```objectivec
- (void)testButtonBackgroundColorForStateFallbackBehavior {
  // Given
  MDCButton *button = [[MDCButton alloc] init];

  // When
  [button setBackgroundColorForState:UIColor.magentaColor forState:UIControlStateNormal];

  // Then
  UIControlState maximalControlState = UIControlStateNormal | UIControlStateSelected |
      UIControlStateDisabled | UIControlStateHighlighted;
  for (NSUInteger state = 0; state <= maximalControlState; ++state) {
      XCTAssertEqualObjects([button backgroundColorForState:state], UIColor.magentaColor);
   }
}
```

### Version number
54.8.0

---
#### Internal data
- Associated internal bug: [b/117178437](http://b/117178437)

### PR DESCRIPTION
**Pull request by romoore**
The rendered (displayed) behavior of background colors in MDCButton is as
expected - when only the `.normal` state is configured, the button keeps that
background at all times.

This is an API-breaking change but does not affect the visual presentation of MDCButton.

Closes #3411

### CODE REVIEW

File: components/Buttons/tests/unit/ButtonsColorThemerTests.m [hunk 1]
```diff
@@ -45,10 +45,11 @@ - (void)testTextButtonColorThemer {
   NSUInteger maximumStateValue = UIControlStateNormal | UIControlStateSelected |
       UIControlStateHighlighted | UIControlStateDisabled;
   for (NSUInteger state = 0; state <= maximumStateValue; ++state) {
-    if ((state == UIControlStateNormal)||(state == UIControlStateDisabled))  {
-      continue; // These two states are manually checked above.
+    if (state == UIControlStateDisabled) {
+      continue;  // This state is manually checked above.
```
Comment (galiak11) [inline components/Buttons/tests/unit/ButtonsColorThemerTests.m:49]: Why is the disabled state skipped?  Doesn't it default to the normal state like the other states?

File: components/Buttons/src/MDCButton.m [hunk 2]
```diff
@@ -546,7 +546,8 @@ - (UIColor *)backgroundColorForState:(UIControlState)state {
   if ((state & UIControlStateHighlighted) == UIControlStateHighlighted) {
     state = state & ~UIControlStateDisabled;
   }
-  return _backgroundColors[@(state)];
+
+  return _backgroundColors[@(state)] ?: _backgroundColors[@(UIControlStateNormal)];
```
Comment (galiak11) [inline components/Buttons/src/MDCButton.m:550]: I wonder if this is correct for the highlighted color. Are we expecting the highlighted background color to be same as the ink color?

**Review by galiak11 — APPROVED** [review]
Why is this an API-breaking change? I don't see any changes to the API.

File: components/Buttons/src/MDCButton.m (hunk 2 above)
Comment (romoore) [reply components/Buttons/src/MDCButton.m:550]: I'm not sure I follow - where did you get the Ink color from?

File: components/Buttons/tests/unit/ButtonsColorThemerTests.m (hunk 1 above)
Comment (romoore) [reply components/Buttons/tests/unit/ButtonsColorThemerTests.m:49]: That's a fair point. I was trying to change the behavior as little as possible - I didn't actually check to see what the `.disabled` was set to.

**Comment by romoore** [comment]
@galiak11 This changes the *behavior* of the API. The `backgroundColorForState:` used to return the value set, or `nil` if none had been explicitly set. It now returns the value that will be *displayed* for that state.  I'm not sure that the behavior was an actual bug, since we did it so often in MDCButton (around half the properties).  That's why I describe it as breaking the API.

File: components/Buttons/tests/unit/ButtonsColorThemerTests.m (hunk 1 above)
Comment (romoore) [reply components/Buttons/tests/unit/ButtonsColorThemerTests.m:49]: The `titleColorForState:` is different for `.disabled` than it is for others. Even though `backgroundColorForState:` doesn't change, the loop is checking both.

**Pull request merged**